raft/leaderElection/leaderelection_test.go

```diff
@@ -0,0 +1,45 @@
+package leaderelection
+
+import (
+	"github.com/SUMUKHA-PK/Raft-Distributed-Consensus/types"
+	"testing"
+	"time"
+)
+
+func TestIncrementTermAndBecomeCandidate(t *testing.T) {
+}
+
+func TestGetTimer(t *testing.T) {
+	minTimeout := 150
+	maxTimeout := 650
+
+	// Test if timer is within expected bounds
+	timer := getTimer(minTimeout, maxTimeout)
+	timerDuration := timer.getDuration()
+
+	if timerDuration < minTimeout*Time.Millsecond {
+		t.Error("Timer runs for less time than expected")
+	} else if timerDuration > maxTimeout*Time.Millisecon {
+		t.Error("Timer runs for more time than expected")
+	}
+
+	// Test if timer is random
```

Comment:
How do I test ranndomness in the durations given by the timer? 

I am thinking of creating 5 timers using same min and max time out, calculate mean and measure absolute difference which should be somewhere around ~15% of duration passed.

Any simpler ideas?

Reply:
Seeing that 15% passed, its a bad idea.
Lets keep this on hold and find new methods for it.
I think "random" , since is a `go` library, mustn't need testing.

Reply:
Now that I think about it - the seeding should belong to main function.

If we use the function to create an array of timers, it does not make sense to seed it everytime. Will shift it